### ISSUE TITLE
[BUG] Error executing 'postInstallation': ENOTEMPTY: directory not empty, rename '/opt/bitnami/jenkins/plugins' -> '/opt/bitnami/jenkins/jenkins_home/plugins'

### ISSUE THREAD

**Issue by madmath03** [bitnami/bitnami-docker-jenkins#57]
**Description**

**Steps to reproduce the issue:**

1. Use the `RUN /install-plugins.sh matrix-auth authorize-project ldap git github gitlab artifactory nexus-jenkins sonar` to add plugins to image (as described in documentation)
2. Build and start the container

**Describe the results you received:**

Startup fails with following error message:
```
Error executing 'postInstallation': ENOTEMPTY: directory not empty, rename '/opt/bitnami/jenkins/plugins' -> '/opt/bitnami/jenkins/jenkins_home/plugins'
```

**Describe the results you expected:**

Adding plugins should not break startup

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
Docker version 18.09.9, build 039a7df9ba
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.24.1, build 4667896b
docker-py version: 3.7.3
CPython version: 3.6.8
OpenSSL version: OpenSSL 1.1.0j  20 Nov 2018
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by marcosbc** [comment]
@madmath03 Unfortunately I could not reproduce your issue. I applied the following change:

```
diff --git a/2/debian-10/Dockerfile b/2/debian-10/Dockerfile
index 8330805..0aecf00 100644
--- a/2/debian-10/Dockerfile
+++ b/2/debian-10/Dockerfile
@@ -31,3 +31,5 @@ EXPOSE 8080 8443 50000

 ENTRYPOINT [ "/app-entrypoint.sh" ]
 CMD [ "/run.sh" ]
+
+RUN /install-plugins.sh matrix-auth authorize-project ldap git github gitlab artifactory nexus-jenkins sonar
```

When starting it works properly:

```
...
jenkins_1  | nami    INFO  Initializing jenkins
jenkins_1  | jenkins INFO  Moving plugins to /opt/bitnami/jenkins/jenkins_home...
jenkins_1  | jenkins INFO  Initializing Jenkins configuration...
jenkins_1  | jenkins INFO  Preparing data to persist...
...
```

Could you share the Docker Image version/tag you're using?

**Comment by marcosbc** [comment]
Also, could you run the commands below and let us know what you get?

```
docker run --rm YOUR_JENKINS_IMAGE ls -la /opt/bitnami/jenkins /opt/bitnami/jenkins/plugins /opt/bitnami/jenkins/jenkins_home /opt/bitnami/jenkins/jenkins_home/plugins /opt/bitnami/jenkins/jenkins_home/plugins/plugins
```

**Comment by madmath03** [comment]
hi @marcosbc 

> Could you share the Docker Image version/tag you're using?

I was testing this with bitnami/jenkins:2-debian-10.
But honestly, the issue is very weird since I got it on my local environment, but I do not get it when building with Travis-CI or Dockerhub...

The docker image is available on my company's github : https://github.com/Monogramm/docker-bitnami-jenkins/tree/master/images/2/debian


> Also, could you run the command below and let us know what you get?

Here it is:
```
Welcome to the Bitnami jenkins container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-jenkins
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-jenkins/issues
Send us your feedback at containers@bitnami.com

ls: cannot access '/opt/bitnami/jenkins/jenkins_home': No such file or directory
ls: cannot access '/opt/bitnami/jenkins/jenkins_home/plugins': No such file or directory
ls: cannot access '/opt/bitnami/jenkins/jenkins_home/plugins/plugins': No such file or directory
/opt/bitnami/jenkins:
total 61952
drwxr-xr-x  3 root    root        4096 Feb 15 02:36 .
drwxrwxr-x 10 bitnami bitnami     4096 Feb 15 02:36 ..
-rw-r--r--  1 root    root           0 Jan 29 16:06 .buildcomplete
-rw-r--r--  1 root    root    63422555 Jan 29 16:06 jenkins.war
drwxr-xr-x  2 root    root        4096 Feb 15 02:36 plugins

/opt/bitnami/jenkins/plugins:
total 63036
drwxr-xr-x 2 root root    4096 Feb 15 02:36 .
drwxr-xr-x 3 root root    4096 Feb 15 02:36 ..
-rw-r--r-- 1 root root 4279042 Jan 29 16:06 ace-editor.jpi
-rw-r--r-- 1 root root   82963 Jan 29 16:06 ant.jpi
-rw-r--r-- 1 root root  107210 Jan 29 16:06 antisamy-markup-formatter.jpi
-rw-r--r-- 1 root root 1762001 Jan 29 16:06 apache-httpcomponents-client-4-api.jpi
-rw-r--r-- 1 root root   14638 Jan 29 16:06 authentication-tokens.jpi
-rw-r--r-- 1 root root 4885133 Jan 29 16:06 bouncycastle-api.jpi
-rw-r--r-- 1 root root  308984 Jan 29 16:06 branch-api.jpi
-rw-r--r-- 1 root root   95728 Jan 29 16:06 build-timeout.jpi
-rw-r--r-- 1 root root  219708 Jan 29 16:06 cloudbees-folder.jpi
-rw-r--r-- 1 root root   80803 Jan 29 16:06 credentials-binding.jpi
-rw-r--r-- 1 root root  971342 Jan 29 16:06 credentials.jpi
-rw-r--r-- 1 root root   27121 Jan 29 16:06 display-url-api.jpi
-rw-r--r-- 1 root root   91018 Jan 29 16:06 docker-commons.jpi
-rw-r--r-- 1 root root   82951 Jan 29 16:06 docker-workflow.jpi
-rw-r--r-- 1 root root 5283231 Jan 29 16:06 durable-task.jpi
-rw-r--r-- 1 root root  796637 Jan 29 16:06 email-ext.jpi
-rw-r--r-- 1 root root   38545 Jan 29 16:06 external-monitor-job.jpi
-rw-r--r-- 1 root root 8503396 Jan 29 16:06 git-client.jpi
-rw-r--r-- 1 root root   30408 Jan 29 16:06 git-server.jpi
-rw-r--r-- 1 root root  903145 Jan 29 16:06 git.jpi
-rw-r--r-- 1 root root 1219480 Jan 29 16:06 github-api.jpi
-rw-r--r-- 1 root root  280664 Jan 29 16:06 github-branch-source.jpi
-rw-r--r-- 1 root root   29682 Jan 29 16:06 github-organization-folder.jpi
-rw-r--r-- 1 root root 2124964 Jan 29 16:06 github.jpi
-rw-r--r-- 1 root root   58363 Jan 29 16:06 gradle.jpi
-rw-r--r-- 1 root root   67784 Jan 29 16:06 handlebars.jpi
-rw-r--r-- 1 root root   24920 Jan 29 16:06 icon-shim.jpi
-rw-r--r-- 1 root root 1972400 Jan 29 16:06 jackson2-api.jpi
-rw-r--r-- 1 root root   33403 Jan 29 16:06 javadoc.jpi
-rw-r--r-- 1 root root   69922 Jan 29 16:06 jdk-tool.jpi
-rw-r--r-- 1 root root  211945 Jan 29 16:06 jquery-detached.jpi
-rw-r--r-- 1 root root  279709 Jan 29 16:06 jsch.jpi
-rw-r--r-- 1 root root  361632 Jan 29 16:06 junit.jpi
-rw-r--r-- 1 root root  130445 Jan 29 16:06 ldap.jpi
-rw-r--r-- 1 root root   74635 Jan 29 16:06 lockable-resources.jpi
-rw-r--r-- 1 root root  126236 Jan 29 16:06 mailer.jpi
-rw-r--r-- 1 root root  396306 Jan 29 16:06 mapdb-api.jpi
-rw-r--r-- 1 root root  128779 Jan 29 16:06 matrix-auth.jpi
-rw-r--r-- 1 root root  230120 Jan 29 16:06 matrix-project.jpi
-rw-r--r-- 1 root root   37051 Jan 29 16:06 momentjs.jpi
-rw-r--r-- 1 root root 1504184 Jan 29 16:06 pam-auth.jpi
-rw-r--r-- 1 root root   34632 Jan 29 16:06 pipeline-build-step.jpi
-rw-r--r-- 1 root root    8479 Jan 29 16:06 pipeline-github-lib.jpi
-rw-r--r-- 1 root root   20613 Jan 29 16:06 pipeline-graph-analysis.jpi
-rw-r--r-- 1 root root   30501 Jan 29 16:06 pipeline-input-step.jpi
-rw-r--r-- 1 root root   18994 Jan 29 16:06 pipeline-milestone-step.jpi
-rw-r--r-- 1 root root 2437756 Jan 29 16:06 pipeline-model-api.jpi
-rw-r--r-- 1 root root    6319 Jan 29 16:06 pipeline-model-declarative-agent.jpi
-rw-r--r-- 1 root root 1491097 Jan 29 16:06 pipeline-model-definition.jpi
-rw-r--r-- 1 root root   38968 Jan 29 16:06 pipeline-model-extensions.jpi
-rw-r--r-- 1 root root  145776 Jan 29 16:06 pipeline-rest-api.jpi
-rw-r--r-- 1 root root   19872 Jan 29 16:06 pipeline-stage-step.jpi
-rw-r--r-- 1 root root   10603 Jan 29 16:06 pipeline-stage-tags-metadata.jpi
-rw-r--r-- 1 root root  201594 Jan 29 16:06 pipeline-stage-view.jpi
-rw-r--r-- 1 root root   17444 Jan 29 16:06 plain-credentials.jpi
-rw-r--r-- 1 root root   21791 Jan 29 16:06 resource-disposer.jpi
-rw-r--r-- 1 root root  218705 Jan 29 16:06 scm-api.jpi
-rw-r--r-- 1 root root 1006025 Jan 29 16:06 script-security.jpi
-rw-r--r-- 1 root root   65885 Jan 29 16:06 ssh-credentials.jpi
-rw-r--r-- 1 root root  179773 Jan 29 16:06 ssh-slaves.jpi
-rw-r--r-- 1 root root   44501 Jan 29 16:06 structs.jpi
-rw-r--r-- 1 root root 7350139 Jan 29 16:06 subversion.jpi
-rw-r--r-- 1 root root   15949 Jan 29 16:06 swarm-3.7.jpi
-rw-r--r-- 1 root root   82494 Jan 29 16:06 timestamper.jpi
-rw-r--r-- 1 root root  654142 Jan 29 16:06 token-macro.jpi
-rw-r--r-- 1 root root  384303 Jan 29 16:06 trilead-api.jpi
-rw-r--r-- 1 root root  470244 Jan 29 16:06 windows-slaves.jpi
-rw-r--r-- 1 root root    8655 Jan 29 16:06 workflow-aggregator.jpi
-rw-r--r-- 1 root root  155980 Jan 29 16:06 workflow-api.jpi
-rw-r--r-- 1 root root  147305 Jan 29 16:06 workflow-basic-steps.jpi
-rw-r--r-- 1 root root 3435811 Jan 29 16:06 workflow-cps-global-lib.jpi
-rw-r--r-- 1 root root  597525 Jan 29 16:06 workflow-cps.jpi
-rw-r--r-- 1 root root   99500 Jan 29 16:06 workflow-durable-task-step.jpi
-rw-r--r-- 1 root root  111555 Jan 29 16:06 workflow-job.jpi
-rw-r--r-- 1 root root   85719 Jan 29 16:06 workflow-multibranch.jpi
-rw-r--r-- 1 root root   15119 Jan 29 16:06 workflow-scm-step.jpi
-rw-r--r-- 1 root root   75642 Jan 29 16:06 workflow-step-api.jpi
-rw-r--r-- 1 root root  380755 Jan 29 16:06 workflow-support.jpi
-rw-r--r-- 1 root root 6349503 Jan 29 16:06 ws-cleanup.jpi
```

**Comment by marcosbc** [comment]
@madmath03 Unfortunately I could not reproduce your issue, even with your image and docker-compose. However, we've identified a potential issue, which fix may resolve the issues on your side. We'll keep you in the loop.

**Comment by marcosbc** [comment]
Hi, it seems we forgot to add an update here. But we wanted to let you know that we made a change recently that may fix this. The image tag was released some time ago, so `bitnami/jenkins:2.204.4-debian-10-r0` already includes the fix.

Note that `2.204.4-debian-10-r3` and later releases include a major change where it is run as a non-root container so you may find issues upgrading. But it includes the fix as well.
